travis.sh

```diff
@@ -6,7 +6,7 @@ for package in $(cat sources.txt) doc/tutorial ; do
   echo testing $package
   pushd $package
   tinc
-  cabal configure --enable-tests --disable-optimization
+  cabal configure --enable-tests --disable-optimization --ghc-options='-Wall -Werror'
```

Comment:
Should `-Wall` be specified here? It looks it's also specified in the individual cabal files. Furthermore, in some cabal files in some places, there are `-fno-warn-...` flags listed. My understanding is that this top-level `-Wall` would override all that. Do we want to make it impossible for components of individual packages to specify `-fno-warn-...` flags?

If yes, then I think they should be removed there. If not, I think we should pass just `-Werror` at this place and leave it to the cabal files what warnings to enable. The latter has my preference.